analyzeme/src/profiling_data.rs

```diff
@@ -82,7 +82,7 @@ impl ProfilingData {
         event_byte_count / RAW_EVENT_SIZE
     }
 
-    pub fn decode_full_event<'a>(&'a self, event_index: usize) -> Event<'a> {
+    fn decode_full_event<'a>(&'a self, event_index: usize) -> Event<'a> {
```

Comment:
Write this line as `pub(crate) fn decode_full_event<'a>(&'a self, event_index: usize) -> Event<'a> {`, then `decode_full_event` can be used from within the entire `analyzeme` crate, but not from outside of the crate. That's a sensible level of visibility for our purposes.

Reply:
Thanks a lot. Will update the code.